apps/indexer/src/api/services/transactions/transactions.service.ts

```diff
@@ -0,0 +1,181 @@
+import {
+  TransactionsRepository,
+  AffectedSupplyFilters,
+  TransactionWithChildren,
+} from "../../repositories/transactions.repository";
+import {
+  TransactionsRequest,
+  TransactionsResponse,
+  TransactionMapper,
+  DBTransaction,
+  DBTransfer,
+  DBDelegation,
+} from "../../mappers/transactions";
+
+type TransactionFilters = {
+  from?: string;
+  to?: string;
+  minAmount?: number;
+  maxAmount?: number;
+  affectedSupplyFilters: AffectedSupplyFilters;
+};
+
+export class TransactionsService {
+  constructor(private repository: TransactionsRepository) {}
```

Comment:
it would be better if you define the TransactionRepository interface right above this class with just the methods it needs